builtin/providers/azurerm/resource_arm_image.go

```diff
@@ -0,0 +1,371 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceArmManagedImage() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmManagedImageCreate,
+		Read:   resourceArmManagedImageRead,
+		Update: resourceArmManagedImageCreate,
+		Delete: resourceArmManagedImageDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"source_virtual_machine_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"os_disk_os_type"},
+			},
+
+			"os_disk_os_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(compute.Linux),
+					string(compute.Windows),
+				}, true),
+			},
+
+			"os_disk_os_state": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(compute.Generalized),
+					string(compute.Specialized),
+				}, true),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+			},
+
+			"os_disk_managed_disk_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"os_disk_blob_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"os_disk_caching": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
```

Comment:
Is there a default value here? Within VM's there's defaults of `ReadWrite`, `ReadOnly` and `None` depending on if it's an OS/Data disk with a Premium/Standard SKU